File: internal/content/testdata/bench/csharp/nc-valid-sudoku.cs
using System.Collections.Generic;

public class Solution {
    public bool is_valid_sudoku(string[][] board) {
        var rows = new HashSet<string>[9];
        var cols = new HashSet<string>[9];
        var boxes = new HashSet<string>[9];
        for (int i = 0; i < 9; i++) {
            rows[i] = new HashSet<string>();
            cols[i] = new HashSet<string>();
            boxes[i] = new HashSet<string>();
        }
        for (int r = 0; r < 9; r++) {
            for (int c = 0; c < 9; c++) {
                string val = board[r][c];
                if (val == ".") continue;
                int boxIdx = (r / 3) * 3 + (c / 3);
                if (!rows[r].Add(val) || !cols[c].Add(val) || !boxes[boxIdx].Add(val))
                    return false;
            }
        }
        return true;
    }
}

File: internal/content/testdata/bench/typescript/nc-valid-sudoku.ts
function is_valid_sudoku(board: string[][]): boolean {
    const rows: Set<string>[] = Array.from({length: 9}, () => new Set<string>());
    const cols: Set<string>[] = Array.from({length: 9}, () => new Set<string>());
    const boxes: Set<string>[] = Array.from({length: 9}, () => new Set<string>());
    for (let r = 0; r < 9; r++) {
        for (let c = 0; c < 9; c++) {
            const val = board[r][c];
            if (val === '.') continue;
            const box_idx = Math.floor(r / 3) * 3 + Math.floor(c / 3);
            if (rows[r].has(val) || cols[c].has(val) || boxes[box_idx].has(val)) {
                return false;
            }
            rows[r].add(val);
            cols[c].add(val);
            boxes[box_idx].add(val);
        }
    }
    return true;
}
